apis/goharbor.io/v1alpha2/harbor_types.go

```diff
@@ -56,8 +58,10 @@ type HarborSpec struct {
 	// +kubebuilder:validation:Optional
 	InternalTLS HarborInternalTLSSpec `json:"internalTLS"`
 
-	// +kubebuilder:validation:Required
-	ImageChartStorage HarborStorageImageChartStorageSpec `json:"imageChartStorage"`
+	// Skip OpenAPI schema validation
+	// Use validating webhook to do verification (field required)
+	// +kubebuilder:validation:Optional
+	ImageChartStorage *HarborStorageImageChartStorageSpec `json:"imageChartStorage,omitempty"`
```

Comment:
Please, do not edit an existing apis version if it has been released.
I suggested to @steven-zou to implements those changes in `v1alpha3`.
About the new CRD cluster, do as you wish: keep it in v1alpha2 or move it to v1alpha3 (I suggest to move it at the same time).

Be careful about keep versions constant as much as possible once released.

PS: I suggest to keep about 3 versions on the master branch, so here keep the v1alpha2 as it is and add the v1alpha3 as needed.

Reply:
@holyhope 

CRD version will be upgraded to v1alpha3, we have an issue https://github.com/goharbor/harbor-operator/issues/397 tracking.

But I want to clarify here, to reduce future maintenance work, I'd like to say the upcoming major release 1.0 should be the start point for considering multiple CRD versions or CRD upgrading cases. Actually, the previous 0.5 and 0.6 have totally different architecture and design. And for those two versions, we also did not announce them as production-ready releases. We'll not consider any upgrading supports for those two 0.x versions.